app/controllers/applications_controller.rb

```diff
@@ -0,0 +1,41 @@
+class ApplicationsController < ApplicationController
+  def show
+    @application = Application.find(params[:id])
+    @pets = @application.pets
+    if params[:search] == "pet" && params[:pet_name].present? 
+      pet_name = params[:pet_name]
+      @search_pets = Pet.where("name ILIKE ?", "%#{pet_name}%")
+    else 
+      @search_pets = []
+    end
+  end
+
+  def new
+  end
+
+  def create
+    application = Application.create({
+      name: params[:name],
+      street_address: params[:street_address],
+      city: params[:city],
+      state: params[:state],
+      zip_code: params[:zip_code],
+      description: params[:description],
+      status: "In Progress"
+    })
+
+    if application.save
```

Comment:
In order for `application.save` to work properly for error handling, line 17 should have Application.new (i.e. you just instantiate the object on line 17 with .new and then .save on line 27 tries to save it to the db)